Cubical/Foundations/HLevels.agda

```diff
@@ -111,16 +114,92 @@ hLevelPi (suc (suc n)) h f g =
 isSetPi : ((x : A) → isSet (B x)) → isSet ((x : A) → B x)
 isSetPi Bset = hLevelPi 2 (λ a → Bset a)
 
+squeezeSq≡
+  : ∀{w x y z : A}
+  → (p : w ≡ y) (q : w ≡ x) (r : y ≡ z) (s : x ≡ z)
+  → (q ≡ p ∙∙ r ∙∙ sym s) ≡ (Square p q r s)
+squeezeSq≡ p q r s k
+  = Square
+      (λ j → p (j ∧ k))
+      q
+      (λ j → doubleCompPath-filler p r (sym s) j (~ k))
+      (λ j → s (j ∧ k))
+
 isSet→isSet' : isSet A → isSet' A
-isSet→isSet' {A = A} Aset {x} {y} {z} {w} p q r s =
-  J (λ (z : A) (r : x ≡ z) → ∀ {w : A} (s : y ≡ w) (p : x ≡ y) (q : z ≡ w) → PathP (λ i → Path A (r i) (s i) ) p q) helper r s p q
-  where
-    helper : ∀ {w : A} (s : y ≡ w) (p : x ≡ y) (q : x ≡ w) → PathP (λ i → Path A x (s i)) p q
-    helper {w} s p q = J (λ (w : A) (s : y ≡ w) → ∀ p q → PathP (λ i → Path A x (s i)) p q) (λ p q → Aset x y p q) s p q
+isSet→isSet' {A = A} Aset {x} {y} {z} {w} p q r s
+  = transport (squeezeSq≡ r p q s) (Aset _ _ p (r ∙∙ q ∙∙ sym s))
 
 isSet'→isSet : isSet' A → isSet A
 isSet'→isSet {A = A} Aset' x y p q = Aset' p q refl refl
 
+squeezeFace≡
+  : ∀{w x y z w' x' y' z' : A}
+  → {p : w ≡ y} {q : w ≡ x} {r : y ≡ z} {s : x ≡ z}
+  → {p' : w' ≡ y'} {q' : w' ≡ x'} {r' : y' ≡ z'} {s' : x' ≡ z'}
+  → {a : w ≡ w'} {b : x ≡ x'} {c : y ≡ y'} {d : z ≡ z'}
+  → (ps : Square a p p' c) (qs : Square a q q' b)
+  → (rs : Square c r r' d) (ss : Square b s s' d)
+  → Square p q r s ≡ Square p' q' r' s'
+squeezeFace≡ ps qs rs ss k = Square (ps k) (qs k) (rs k) (ss k)
+
+squeezeCu≡
+  : ∀{w x y z w' x' y' z' : A}
+  → {p : w ≡ y} {q : w ≡ x} {r : y ≡ z} {s : x ≡ z}
+  → {p' : w' ≡ y'} {q' : w' ≡ x'} {r' : y' ≡ z'} {s' : x' ≡ z'}
+  → {a : w ≡ w'} {b : x ≡ x'} {c : y ≡ y'} {d : z ≡ z'}
+  → (ps : Square a p p' c) (qs : Square a q q' b)
+  → (rs : Square c r r' d) (ss : Square b s s' d)
+  → (f0 : Square p q r s) (f1 : Square p' q' r' s')
+  → (f0 ≡ transport⁻ (squeezeFace≡ ps qs rs ss) f1) ≡ Cube ps qs rs ss f0 f1
+squeezeCu≡ ps qs rs ss f0 f1 τ
+  = Cube
+      (λ j → ps (j ∧ τ))
+      (λ j → qs (j ∧ τ))
+      (λ j → rs (j ∧ τ))
+      (λ j → ss (j ∧ τ))
+      f0
+      (toPathP {A = F} (transportTransport⁻ (squeezeFace≡ ps qs rs ss) f1) τ)
+  where
+  F : I → Set _
+  F k = Square (ps k) (qs k) (rs k) (ss k)
+
+transposeSq
+  : ∀{w x y z : A} {p : w ≡ y} {q : w ≡ x} {r : y ≡ z} {s : x ≡ z}
+  → Square p q r s
+  → Square q p s r
+transposeSq sq i j = sq j i
+
+transposeCu
```

Comment:
I don't think we need this as a separate lemma

Reply:
You're right, it was easy enough to inline these. I'm not sure I could have discovered the inlining without the separate definitions, though. :)